tests/test_workers/test_tasks/test_rubygems.py

```diff
@@ -70,3 +79,126 @@ def test_get_config_file(tmp_path, exists):
+@mock.patch("cachito.workers.tasks.rubygems.set_request_state")
+@mock.patch("cachito.workers.tasks.utils.get_request_state")
+@mock.patch("cachito.workers.tasks.rubygems.get_request")
+def test_fetch_rubygems_source(
+    mock_get_request,
+    mock_get_state,
+    mock_set_state,
+    mock_update_cfg,
+    mock_prepare_nexus,
+    mock_finalize_nexus,
+    mock_resolve,
+    mock_cert,
+    mock_get_url,
+    package_subpath,
+    with_cert,
+    tmpdir,
+):
+    # Setup
+    pkg_data = {
+        "package": {
+            "name": "pkg_name",
+            "version": "1.0.0",
+            "type": "rubygems",
+            "path": package_subpath,
+        },
+        "dependencies": [
+            {
+                "kind": "GEM",
+                "name": "ci_reporter",
+                "version": "2.0.0",
+                "path": "some/path",
+                "type": "rubygems",
+            },
+        ],
+    }
+    request = {"id": 1}
+    password = "password"
+
+    mock_get_state.return_value = "in_progress"
+    mock_resolve.return_value = pkg_data
+    mock_finalize_nexus.return_value = password
+    mock_get_request.return_value = request
+    stub_url = "stub_url"
+    mock_get_url.return_value = stub_url
+
+    if package_subpath:
+        package_configs = [{"path": package_subpath}]
+    else:
+        package_configs = None
+
+    if package_subpath is None or package_subpath == ".":
+        bundle_config_path = "app/.bundle/config"
+        nexus_ca_path = "rubygems-proxy-ca.pem"
+    else:
+        bundle_config_path = f"app/{package_subpath}/.bundle/config"
+        nexus_ca_path = "../../rubygems-proxy-ca.pem"
+
+    cfg = []
+    cert_contents = "stub_cert"
+    if with_cert:
+        mock_cert.return_value = cert_contents
+        b64_cert_contents = base64.b64encode(cert_contents.encode()).decode()
+        cfg.append(
+            {"content": b64_cert_contents, "path": "app/rubygems-proxy-ca.pem", "type": "base64"}
+        )
+    else:
+        mock_cert.return_value = None
+
+    bundle_config = get_bundle_base_config(stub_url)
+    if with_cert:
+        bundle_config += f"\nBUNDLE_SSL_CA_CERT: {nexus_ca_path}"
+    b64_bundle_config = base64.b64encode(bundle_config.encode()).decode()
+    cfg.append({"content": b64_bundle_config, "path": bundle_config_path, "type": "base64"})
+
+    # Exercise
+    rubygems.fetch_rubygems_source(request["id"], package_configs=package_configs)
+
+    # Verify
+    mock_prepare_nexus.assert_called_once()
+    mock_finalize_nexus.assert_called_once()
+
+    calls = call(request["id"], cfg)
+    assert calls == mock_update_cfg.call_args
+
+    expected = pkg_data["package"].copy()
+    del expected["path"]
+    expected_dependencies = pkg_data["dependencies"]
+    del expected_dependencies[0]["path"]
+    expected["dependencies"] = expected_dependencies
+    if package_subpath and package_subpath != os.curdir:
+        expected["path"] = package_subpath
```

Comment:
nitpick: maybe delete `expected["path"]` only when needed? Rather than always deleting it and sometimes restoring it